src/transforms/remap.rs

```diff
@@ -106,11 +112,45 @@ impl Remap {
     const fn runtime(&self) -> &Runtime {
         &self.runtime
     }
+
+    fn annotate_dropped(&self, event: &mut Event, reason: &str, error: ExpressionError) {
+        match event {
+            Event::Log(ref mut log) => {
+                log.insert(
+                    log_schema().metadata_key(),
+                    serde_json::json!({
+                        "dropped": {
+                            "reason": reason,
+                            "message": error.to_string(),
+                            "component_id": self.component_key,
+                            "component_type": "remap",
+                            "component_kind": "transform",
+                        }
+                    }),
+                );
+            }
+            Event::Metric(ref mut metric) => {
+                let m = log_schema().metadata_key();
+                metric.insert_tag(format!("{}.dropped.reason", m), reason.into());
+                metric.insert_tag(format!("{}.dropped.message", m), error.to_string());
```

Comment:
I'd say we could probably drop message since it'd make an awkward metrics tag. We can add it later if users ask for it.